opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -1467,6 +1469,36 @@ checkGroupConstraintsProd(const std::string& name,
     const auto chain = groupChainTopBot(name, group.name(), schedule, reportStepIdx);
     // Because 'name' is the last of the elements, and not an ancestor, we subtract one below.
     const std::size_t num_ancestors = chain.size() - 1;
+
+    // check whether guide rate is violated
+    if (check_guide_rate) {
+        for (std::size_t ii = 1; ii < num_ancestors; ++ii) {
+            if (guideRate->has(chain[ii])) {
+                const auto& guided_group = chain[ii];
+                const Scalar grefficiency
+                            = schedule.getGroup(guided_group, reportStepIdx).getGroupEfficiencyFactor();
+                const Scalar currentRateFraction = grefficiency * localCurrentRate(guided_group) / (localCurrentRate(chain[ii-1]));
+                const Scalar guiderateFraction = localFraction(guided_group);
+                // we add a factor here to avoid switching due to numerical instability
+                const Scalar factor = 1.01;
+                if (currentRateFraction > (guiderateFraction * factor)) {
+                    return std::make_pair(true, guiderateFraction / currentRateFraction);
+                }
+            }
+        }
+    }
+
+    if (schedule.hasWell(name) && wellState.well(name).group_target) { // for wells we already have computed the target
+        const Scalar current_well_rate_available = -tcalc.calcModeRateFromRates(rates); // Switch sign since 'rates' are negative for producers.
+        Scalar group_target_rate_available = *wellState.well(name).group_target;
```

Comment:
This should be const, also on lines 1538, 1847 and 1886. This used to be const, I assume the change was unintentional.